packages/emitter-framework/test/typescript/components/union-declaration.test.tsx

```diff
@@ -1,182 +1,379 @@
+              meow: boolean;
+            }
 
-        const testFile = res.contents.find((file) => file.path === "test.ts");
-        assert(testFile, "test.ts file not rendered");
-        const actualContent = await format(testFile.contents as string, { parser: "typescript" });
-        const expectedContent = await format(`let x:"one" | "two" = "one"`, {
-          parser: "typescript",
+            @test model Dog {
+              name: string;
+              bark: boolean;
+            }
+
+            @discriminated
+            @test union Pet {
+              cat: Cat,
+              dog: Dog,
+            }
+          `)) as { Pet: Union; Cat: Model; Dog: Model };
+
+        const res = render(
+          <Output program={runner.program}>
+            <SourceFile path="test.ts">
+              <InterfaceDeclaration type={Cat} />
+              <hbr />
+              <InterfaceDeclaration type={Dog} />
+              <hbr />
+              <UnionDeclaration type={Pet} />
+            </SourceFile>
+          </Output>,
+        );
+        assertFileContents(
+          res,
+          d`
+            interface Cat {
+              name: string;
+              meow: boolean;
+            }
+            interface Dog {
+              name: string;
+              bark: boolean;
+            }
+            type Pet = {
+              kind: "cat";
+              value: Cat;
+            } | {
+              kind: "dog";
+              value: Dog;
+            };
+          `,
+        );
+      });
+
+      describe("Discriminated Union with no envelope", () => {
+        it("renders named discriminated union", async () => {
+          const { Pet, Cat, Dog } = (await runner.compile(`
+            namespace DemoService;
+
+            @test model Cat {
+              name: string;
+              meow: boolean;
+            }
+
+            @test model Dog {
+              name: string;
+              bark: boolean;
+            }
+
+            @discriminated(#{ envelope: "none", discriminatorPropertyName: "dataKind" })
+            @test union Pet {
+              cat: Cat,
+              dog: Dog,
+            }
+          `)) as { Pet: Union; Cat: Model; Dog: Model };
+
+          const res = render(
+            <Output program={runner.program}>
+              <SourceFile path="test.ts">
+                <InterfaceDeclaration type={Cat} />
+                <hbr />
+                <InterfaceDeclaration type={Dog} />
+                <hbr />
+                <UnionDeclaration type={Pet} />
+              </SourceFile>
+            </Output>,
+          );
+
+          assertFileContents(
+            res,
+            d`
+              interface Cat {
+                name: string;
+                meow: boolean;
+              }
+              interface Dog {
+                name: string;
+                bark: boolean;
+              }
+              type Pet = {
+                dataKind: "cat"
+              } & Cat | {
```

Comment:
If you were feeling extra you could put parens around each variant here. Prettier does so just because we humans aren't great at recognizing operator precedence. Not required.